openbox/frame.h

```diff
@@ -59,9 +59,9 @@ typedef enum {
 
 #define FRAME_CONTEXT(co, cl) ((cl && cl->type != OB_CLIENT_TYPE_DESKTOP) ? \
                                co == OB_FRAME_CONTEXT_FRAME : FALSE)
-#define CLIENT_CONTEXT(co, cl) ((cl && cl->type == OB_CLIENT_TYPE_DESKTOP) ? \
+#define CLIENT_CONTEXT(co, cl) ((cl && (cl->type == OB_CLIENT_TYPE_DESKTOP) ? \
```

Comment:
This should return a context, not a boolean, now it can return FALSE, DESKTOP or CLIENT, which seems incorrect?

Reply:
Actually, it's always a boolean. See, the result of the ?: operator is either 'co == OB_FRAME_CONTEXT_DESKTOP' which evaluates as a boolean, 'co == OB_FRAME_CONTEXT_CLIENT', which is also a boolean.
However, I'm double checking because I'm now unsure it fixes what I intended to fix.

Reply:
OK, I double checked. The situation was the following:
- In openbox/mouse.c:267, it is checked if we're in client context to decide whether to fall through for inserting a button release event.
- When client is NULL, we get the same result as to when cl->type != OB_CLIENT_DESKTOP, that is 'co == OB_FRAME_CONTEXT_DESKTOP'.
However, if client == NULL, that doesn't seem correct, as we said there is no client to begin with.
I may be reading this the wrong way, but that's how it seems to me.